### PR TITLE
[PWGCF] New task for TwoParticleCorrelation

### PR DESCRIPTION
**Pull request by Thorkj**
I made a task adding the deltaPhi and deltaEta along with, Nch, pT, to a sparse for local processing.

### CODE REVIEW

File: PWGCF/TwoParticleCorrelations/Tasks/CorrSparse.cxx [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+        }
+
+        float deltaPhi = track1.phi() - track2.phi();
+        float deltaEta = track1.eta() - track2.eta();
+        if (deltaPhi > 1.5f * PI) {
+          deltaPhi -= TwoPI;
+        }
+        if (deltaPhi < -PIHalf) {
+          deltaPhi += TwoPI;
+        }
```
Comment (vkucera) [inline PWGCF/TwoParticleCorrelations/Tasks/CorrSparse.cxx:115]: Do not duplicate already available utilities. See `RecoDecay::constrainAngle`.

File: PWGCF/TwoParticleCorrelations/Tasks/CorrSparse.cxx [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+    return true;
+  }
+
+  template <typename TTracks>
+  void fillCorrelations(TTracks tracks1, TTracks tracks2, float posZ, int system, float Nch) // function to fill the Output functions (sparse) and the delta eta and delta phi histograms
+  {
+    // loop over all tracks
+    for (auto& track1 : tracks1) {
+
+      for (auto& track2 : tracks2) {
```
Comment (vkucera) [inline PWGCF/TwoParticleCorrelations/Tasks/CorrSparse.cxx:114]:
```suggestion
    for (auto const& track1 : tracks1) {

      for (auto const& track2 : tracks2) {
```

File: PWGCF/TwoParticleCorrelations/Tasks/CorrSparse.cxx [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+#include "Common/DataModel/Centrality.h"
+#include "PWGCF/Core/CorrelationContainer.h"
+#include "PWGCF/Core/PairCuts.h"
+
+#include "TMath.h"
+#include "TList.h"
+#include "TProfile.h"
+#include "TRandom3.h"
+#include <cmath>
+#include <iostream>
```
Comment (vkucera) [inline PWGCF/TwoParticleCorrelations/Tasks/CorrSparse.cxx:30]: Where do you need these headers?

File: PWGCF/TwoParticleCorrelations/Tasks/CorrSparse.cxx [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+using Hash = Hashes::iterator;
+} // namespace o2::aod
+
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::framework::expressions;
+
+int s = 1; // same event
+int m = 2; // mixed event
+
```
[the remark this answers is not in the document]
Comment (Thorkj) [reply PWGCF/TwoParticleCorrelations/Tasks/CorrSparse.cxx:50]: Sorry for the late response, I have now removed them as I didn't use them.

File: PWGCF/TwoParticleCorrelations/Tasks/CorrSparse.cxx [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+namespace o2::aod
+{
+namespace hash
+{
+DECLARE_SOA_COLUMN(Bin, bin, int);
+} // namespace hash
+DECLARE_SOA_TABLE(Hashes, "AOD", "HASH", hash::Bin);
+
+using Hash = Hashes::iterator;
+} // namespace o2::aod
```
Comment (victor-gonzalez) [inline PWGCF/TwoParticleCorrelations/Tasks/CorrSparse.cxx:35]: Please, remove it as it is not used

File: PWGCF/TwoParticleCorrelations/Tasks/CorrSparse.cxx (hunk 1 above)
Comment (victor-gonzalez) [reply PWGCF/TwoParticleCorrelations/Tasks/CorrSparse.cxx:114]: You can find it on https://github.com/AliceO2Group/O2Physics/blob/master/Common/Core/RecoDecay.h

**Review by victor-gonzalez — CHANGES_REQUESTED** [review]

File: PWGCF/TwoParticleCorrelations/Tasks/CorrSparse.cxx (hunk 5 above)
Comment (Thorkj) [reply PWGCF/TwoParticleCorrelations/Tasks/CorrSparse.cxx]: I have now removed it

**Review by victor-gonzalez — CHANGES_REQUESTED** [review]
Please, implement the first suggestion from Vit which I expanded giving you where you can find what he suggested

**Review by victor-gonzalez — APPROVED** [review]
Thanks for implementing our suggestions!

**Pull request merged**